### PR TITLE
Set prepend_bos to false by default for Bloom model family

### LINKED ISSUE #774: [Proposal] prepend_bos should by default be set to false for the Bloom model family
### Proposal 

When loading the bloom models, prepend_bos should be set to false by default instead of true.

### Motivation

By default TransformerLens sets prepend_bos to true. However, for the bloom model family setting prepend_bos to true yields completely insensible output which is highly different from the HuggingFace implementation. Turning prepend_bos to false solves this issue (partially, use_kv_cache also needs to be set to false, I will create an issue about this in the future). One should be able to run a model in TransformerLens without knowing all the special options and parameters that need to be set, so in case of the bloom model family prepend_bos should be set to false by default.

I tested this on Google Colab (TL installed with pip) and my Mac (macOS Sequoia 15.0.1) with python 3.10.15 and they both yield the same results

I attached a screenshot of an
Bildschirmfoto 2024-11-08 um 04 16 28
Bildschirmfoto 2024-11-08 um 04 15 58
 example output which shows the difference between setting prepend_bos to true and setting it to false.

- [x] I have checked that there is no similar [issue](https://github.com/TransformerLensOrg/Transformerlens/issues) in the repo (**required**)

### PR DESCRIPTION
**Pull request by degenfabian**
# Description

This PR addresses Issue #774. Bloom models output insensible output when the prepend_bos flag is set to True, so for usability this flag should be set to False by default for the Bloom model family. I'm happy about ideas to implement this more cleanly, because with the current implementation the flag can practically not be set to True anymore, unless it is overwritten when specifically calling a function like generate. I think that's an unpretty solution but since I can't see a use case in which it would make sense to set the flag to True I think it's not terrible, but again very happy for suggestions.

Fixes #774 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### Screenshots
For Screenshots of the difference in the generation of the models please see Issue #774.

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have not rewritten tests relating to key interfaces which would affect backward compatibility

### CODE REVIEW

**Comment by degenfabian** [comment]
Interestingly, adding this if clause makes the acceptance test fail although it fixes the outputs being totally off. I will investigate this further tomorrow.

**Comment by degenfabian** [comment]
Interestingly, I just ran the hugging face implementation on Google Colab to see if I would get a different value than on my Mac, and indeed I did. The loss value I got from Colab now matches what we get from TransformerLens with prepend_bos set to false close enough to be within the error margin, so hooray I guess. I'm slightly confused by this, does anyone have an idea as to why I would get a different loss from the hugging face implementation on my Mac then on Google Colab?

**Comment by degenfabian** [comment]
I just tested this with gpt2-small, and here the output with prepend_bos set to True is not completely unusable but still deviates from Huggingface. If we just want to match HuggingFace as closely as possible, I could imagine adding this in loading_from_pretrained.py and setting the prepend_bos model accordingly for each model we load from Huggingface. I would be happy to do that and rename this PR and issue, what do you think @bryce13950 ?

**Comment by bryce13950** [comment]
Let's talk about this a bit to get up to speed, and figure out if we can find where the divergence is.

File: tests/acceptance/test_hooked_transformer.py [hunk 1]
```diff
@@ -83,6 +83,8 @@ def test_model(name, expected_loss):
     # Runs the model on short text and checks if the loss is as expected
     model = HookedTransformer.from_pretrained(name)
     loss = model(text, return_type="loss")
+    print(4e-5)
+    print(loss.item() - expected_loss)
```
Comment (bryce13950) [inline tests/acceptance/test_hooked_transformer.py:87]: Make sure to remove log points when you submit your PR!

**Comment by bryce13950** [comment]
This looks good besides the log points! Once those are removed, we can merge this

File: tests/acceptance/test_hooked_transformer.py (hunk 1 above)
Comment (degenfabian) [reply tests/acceptance/test_hooked_transformer.py]: So sorry, I forgot about those. I just removed them!
